embulk-core/src/main/java/org/embulk/spi/json/JsonParser.java

```diff
@@ -45,14 +55,19 @@ private static String sampleJsonString(String json) {
     }
 
     private static class StreamParseContext extends AbstractParseContext implements Stream {
-        public StreamParseContext(JsonFactory factory, InputStream in) throws IOException, JsonParseException {
-            super(createParser(factory, in));
+        public StreamParseContext(JsonFactory factory, InputStream in, String pointer) throws IOException {
+            super(createParser(factory, in, pointer));
         }
 
-        private static com.fasterxml.jackson.core.JsonParser createParser(JsonFactory factory, InputStream in)
-                throws IOException {
+        private static com.fasterxml.jackson.core.JsonParser createParser(
+                JsonFactory factory, InputStream in, String pointer
+        ) throws IOException {
             try {
-                return factory.createParser(in);
+                final com.fasterxml.jackson.core.JsonParser baseParser = factory.createParser(in);
+                if (pointer == null || pointer.isEmpty()) {
```

Comment:
We have no reasons to hesitate using `Optional`, especially in `private` regions.

Reply:
Fixed in e9f4f9d